ddtrace/tracer/option.go

```diff
@@ -168,6 +168,19 @@ func (c *config) HasFeature(f string) bool {
 	return ok
 }
 
+func (c *config) DumpForTestAgent() string {
+	envVars := map[string]string{
+		"DD_SERVICE":                             c.serviceName,
+		"DD_TRACE_SPAN_ATTRIBUTE_SCHEMA":         fmt.Sprintf("v%d", c.spanAttributeSchemaVersion),
+		"DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED": strconv.FormatBool(c.peerServiceDefaultsEnabled),
```

Comment:
Passing the config values as env vars is misleading, what if they were provided in-code and not via env vars? Are you sure you don't want to use the values from `os.Getenv` here?

Reply:
These are included within the transport header as `X-Datadog-Trace-Env-Variables`, as a string of mapped key value pairs. We need this in order to have the Datadog tracer configuration at the time of the trace's creation.